server/src/useCases/report/productTestCases/handleGetProductDetailsRequest.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable object-curly-newline */
+/* eslint-disable no-console */
+
+module.exports = class ReportGetProductDetailsRequestHandler {
+  constructor(request, response) {
+    this.req = request;
+    this.res = response;
+    this.dbConnect = request.app.locals.reportCollection;
+  }
+
+  async getProductDetails(repository) {
+    const { limit, offset } = this.req.query;
+    const { id } = this.req.params;
+
+    repository.aggregateByProductId({ id }, async (err, result) => {
+      if (err) {
+        console.error(err);
+        this.res
+          .status(500)
+          .send(`Failed to fetch detailed report summary. Details: \n\t${err}\nPlease contact administrator.`);
```

Comment:
Don't think we need 'Please contact administrator' in the stacktrace. Or is this intended for the end user?